pytorch/cppbuild.sh

```diff
@@ -22,7 +22,7 @@ if [[ "$EXTENSION" == *gpu ]]; then
     export USE_CUDNN=1
     export USE_FAST_NVCC=0
     export CUDA_SEPARABLE_COMPILATION=OFF
-    export TORCH_CUDA_ARCH_LIST="5.0+PTX"
+    export TORCH_CUDA_ARCH_LIST="7.5+PTX"
```

Comment:
Please revert that back to at least 5.0+PTX. This won't work on older cards

Reply:
We must make a tradeoff between support of older cards and support for new computation features. For instance support for half floats won't work (I think) with 5.0+PTX. I thought in 2023 not much people will try to use these presets with pre-turing cards, but I may be wrong.

Reply:
Like I keep telling you, users still have the choice to use LibTorch. 

In any case, like I also told you many times now, please stop trying to do everything at the same time. Revert that and anything else I tell you to, and let's keep discussing that and the rest of your laundry list in future pull requests. Please keep them as short as possible!

Reply:
I will revert. Anyway any serious use of the presets will probably need to use libtorch or custom recompiling.

But this PR is like a major release or rework of the presets. if you don't like that or don't trust me for maintaining it, feel free to not merge it, or merge some features one by one, and I will keep this version on my fork.

Reply:
No, that's not the point, the point is to do one thing at a time. If we want to change the default CUDA archs, it's not only about PyTorch, it's also about OpenCV, ONNX Runtime, and TVM. These projects don't care about interoperability of their libraries, but I do. If you also don't care about using OpenCV, ONNX Runtime, TVM, and PyTorch in the same application, then you should do like @Craigacp who works upstream in the ONNX Runtime repository and publishes binaries that don't work well with OpenCV, etc, and work upstream in the PyTorch repository to work on Java bindings there, and do the same, yes.

Reply:
It looks like it's able to build within 6 hours with "5.0;6.0;7.0+PTX":
https://github.com/bytedeco/javacpp-presets/actions/runs/5255778236/jobs/9496144498
This means any GPU with CC 5.x, 6.x, or 7.x can load the binaries, while newer ones with CC 8.x and 9.x need to compile from PTX. Is that good enough for you?

Reply:
I'm using custom build for my needs anyway. The question is what's the best for users, not for me.

It's a tradeoff between available CC, startup time, binary size, and old hardware support .
You also talked about interoperability, I'm not sure to understand how it's related. 

I'd say binary size is not important so "5.0;6.0;7.0+PTX" is better than "5.0+PTX". 7.5 introduces support for Half (FP16) which is important nowadays so I'd try to include 7.5. Maybe with "5.0+PTX;7.5+PTX" ?

Reply:
CC 7.0 looks like it supports FP16, just not INT8:
https://docs.nvidia.com/deeplearning/tensorrt/support-matrix/index.html#hardware-precision-matrix

Reply:
Ok. 7.5 adds some FP16 features but maybe they are not that relevant. What we can also do is support old hardware with 5.0+PTX and add binaries for the cards we believe are most common at the day of the release, so that those cards get both fast startup and full features. Something like "5.0+PTX;8.6;8.9".

Reply:
If you want to do research for that, please do and open an issue with your findings.

Reply:
We probably just need to refresh the cache. BTW, it looks like NVCC is supported by ccache on Windows as well these days: https://ccache.dev/platform-compiler-language-support.html

If you want to give that a try, we could compile for any number of CUDA archs with build times of 10+ hours and offer 2+ GB binaries to users who want them.

Reply:
By simply relaunching the job until it's complete ? Do you know a way to automate this ?

And how would you handle "to users who want them" ? Shall we create multiple extensions ? "-gpu-ptx", "-gpu-cubins" ?